components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+        compute_w0_and_rho(team, omega, T_mid, p_mid, icol, top_lev, nlev,
+                           // output
+                           w0, rho);
+      });
+}
+
+void compute_values_at_interfaces(haero::ThreadTeamPolicy team_policy,
+                                  const MAMAci::const_view_2d var_mid,
+                                  const MAMAci::view_2d dz, const int nlev_,
+                                  // output
+                                  MAMAci::view_2d var_int) {
+  using CO = scream::ColumnOps<DefaultDevice, Real>;
+
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+
+        const auto var_mid_col = ekat::subview(var_mid, icol);
+        const auto var_int_col = ekat::subview(var_int, icol);
+        const auto dz_col      = ekat::subview(dz, icol);
+
+        const Real bc_top = var_mid_col(0);
+        const Real bc_bot = var_mid_col(nlev_ - 1);
+
+        CO::compute_interface_values_linear(team, nlev_, var_mid_col, dz_col,
+                                            bc_top, bc_bot, var_int_col);
+      });
+}
+
+KOKKOS_INLINE_FUNCTION
+void compute_tke_using_w_sec(const haero::ThreadTeam &team,
+                             const MAMAci::const_view_2d w_sec, const int icol,
+                             const int nlev,
+                             // output
+                             MAMAci::view_2d tke) {
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 0u, nlev + 1),
+      KOKKOS_LAMBDA(int kk) { tke(icol, kk) = (3.0 / 2.0) * w_sec(icol, kk); });
+}
+void compute_tke_using_w_sec(haero::ThreadTeamPolicy team_policy,
+                             const MAMAci::const_view_2d w_sec, const int nlev,
+                             // output
+                             MAMAci::view_2d tke) {
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        compute_tke_using_w_sec(team, w_sec, icol, nlev,
+                                // output
+                                tke);
+      });
+}
+KOKKOS_INLINE_FUNCTION
+void compute_subgrid_scale_velocities(
+    const haero::ThreadTeam &team, const MAMAci::const_view_2d tke,
+    const Real wsubmin, const int icol, const int top_lev, const int nlev,
+    // output
+    MAMAci::view_2d wsub, MAMAci::view_2d wsubice, MAMAci::view_2d wsig) {
+  // More refined computation of sub-grid vertical velocity
+  // Set to be zero at the surface by initialization.
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 0u, top_lev), KOKKOS_LAMBDA(int kk) {
+        wsub(icol, kk)    = wsubmin;
+        wsubice(icol, kk) = 0.001;
+        wsig(icol, kk)    = 0.001;
+      });
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, top_lev, nlev), KOKKOS_LAMBDA(int kk) {
+        wsub(icol, kk) = haero::sqrt(0.5 * (tke(icol, kk) + tke(icol, kk + 1)) *
```

Comment:
Is the "tke" here at the interface (interpolated using mid-layer TKE from SHOC)?

Reply:
Yes, `tke` is assumed to be at interfaces here. The level loop here goes from `top_lev` to `nlev` and we use `kk+1` index of `tke` to fetch data from the last interface level.